parsons/zoom/zoom.py

```diff
@@ -275,3 +275,103 @@ def get_webinar_registrants(self, webinar_id):
         tbl = self._get_request(f"webinars/{webinar_id}/registrants", "registrants")
         logger.info(f"Retrieved {tbl.num_rows} webinar registrants.")
         return tbl
+
+    def get_meeting_poll(self, meeting_id: str, poll_id: str) -> Table:
+        """
+        Get a specific poll for a given meeting ID
+
+        `Args`:
+            meeting_id: str
+                Unique identifier for Zoom meeting
+            poll_id: str
+                Unique identifier for poll
+
+        `Returns`:
+            Parsons Table of all polling responses
+        """
+
+        endpoint = f"meetings/{meeting_id}/polls/{poll_id}"
+        tbl = self._get_request(endpoint=endpoint, data_key="questions")
+
+        if type(tbl) == dict:
+            logger.debug(f"No poll data returned for poll ID {poll_id}")
+            return Table(tbl)
+
+        logger.info(
+            f"Retrieved {tbl.num_rows} rows of metadata [meeting={meeting_id} poll={poll_id}]"
+        )
+
+        return tbl
+
+    def get_meeting_all_polls(self, meeting_id: str) -> Table:
+        """
+        Get all polls for a given meeting ID
+
+        `Args`:
+            meeting_id: str
+                Unique identifier for Zoom meeting
+
+        `Returns`:
+            Parsons Table of all polling responses
+        """
+
+        endpoint = f"meetings/{meeting_id}/polls"
```

Comment:
This looks light to me, but it doesn't seem to be working. I'm running this script (The account ID is a properly named ENV VAR so I don't need to reference it here):

``` python
from parsons.zoom import Zoom
import os

ZOOM_CLIENT_ID = os.environ["ZOOM_CREDENTIAL_API_USERNAME"]
ZOOM_CLIENT_SECRET = os.environ["ZOOM_CREDENTIAL_API_PASSWORD"]

MEETING_ID = 83783398800


def main():
    zoom = Zoom(client_id=ZOOM_CLIENT_ID, client_secret=ZOOM_CLIENT_SECRET)
    print(zoom.get_past_meeting(MEETING_ID))
    print(zoom.get_meeting_all_polls(MEETING_ID))


if __name__ == "__main__":
    main()
```

The first print, getting the meeting details, is working. So I know that the meeting does exist. But the second print, trying to get all of the polls, is failing. 

```
requests.exceptions.HTTPError: HTTP error occurred (404): Not Found, json: {'code': 3001, 'message': 'Meeting does not exist: 83783398800.'}
```

Reply:
This is a great flag - seems like Zoom's REST API has a whole different endpoint for "past" meeting polls:
https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/listPastMeetingPolls

Reply:
For the sake of consistency / verbosity I suppose I'll just write new functions for `past meetings` and `past webinars`?

Reply:
my guess is that if you run `.get_meeting(MEETING_ID)` (instead of `get_past_meeting()`)  it'll throw an exception too, is that right?

Reply:
ack, I should I have caught that that's my bad. Maybe I'll change the existing functions to `get_meeting_all_polls_metadata` and then the `get_meeting_poll` would return data from the reports endpoint? What do you think about that?

Trying to imagine when you'd want the metadata vs. the actual results, I reckon the latter is much more common but I could be wrong

Reply:
That sounds good to me. I don't know when you would want the metadata, but we might as well offer the option!

Reply:
That works! I was able to get poll results. I have two pieces of feedback on it now. 

1) Could we mention in the documentation that you need the report:read:admin scope in your Zoom authentication setup for any of the functions that get reports?

2) I think the return type _could_ be improved, maybe. Here is an example row:

```
{'name': 'John Doe', 'email': 'johndoe@gmail.com', 'question_details': [{'question': 'Do you think this could help you?', 'answer': 'Yes', 'polling_id': 'abcde', 'date_time': '2023-01-20 01:15:50'}, {'question': 'Do you have an upcoming event?', 'answer': 'No', 'polling_id': 'efgh', 'date_time': '2023-01-20 01:16:50'}], 'first_name': 'John', 'last_name': 'Doe'}
```

The question_details column would need some manual unpacking into some kind of _xf table or something.

@shaunagm do you have thoughts on this? Should the Parsons connector focus on shaping API data to make it easier to load into a database? Or should it focus on providing more direct access to the API data?